firedrake/mesh.py

```diff
@@ -224,6 +224,36 @@ def facet_cell_map(self):
                        "facet_to_cell_map")
 
 
+@PETSc.Log.EventDecorator()
+def _from_netgen(ngmesh,comm=None):
+    """
+    Create a DMPlex from an Netgen mesh
+
+    :arg ngmesh: NetGen Mesh
+    """
+    if len(ngmesh.Elements3D()) != 0:
```

Comment:
I only thought about the 2D implementation, but I'll have a look also to the 3D case. There should be no major difference as you were pointing out.
I've never tried to do a 1D mesh in Netgen, but I think they are supported. Do you think it would be useful to have 1D mesh implemented?

Reply:
@dham I think I've implemented all yours and @ksagiyam suggestions for the 2D case, I'll try to have a look at the 3D case soon.

Reply:
Added 3D support but it requires a function from PETSc to be wrapped in petsc4py, I've done it in [this fork](https://gitlab.com/UZerbinati1/petsc/-/tree/uzerbianti/petsc4py).